src/Akka.Persistence.Sql.Linq2Db/persistence.conf

```diff
@@ -89,102 +101,156 @@
       # If set to null, the default `System.Object` serializer is used.
       serializer = null
       
-      tables.journal {
-        
-        #if delete-compatibility-mode is true, both tables are created
-        #if delete-compatibility-mode is false, only journal table will be created.
-        auto-init = true
-
-        #if true, a warning will be logged
-        #if auto-init of tables fails.
-        #set to false if you don't want this warning logged
-        #perhaps if running CI tests or similar.
-        warn-on-auto-init-fail = true
-
-        table-name = "journal"
-        metadata-table-name = "journal_metadata"
-
+      # Default table name and column name mapping
+      # Use this if you're not migrating from old Akka.Persistence plugins
+      default {
         #If you want to specify a schema for your tables, you can do so here.
         schema-name = null
 
-        column-names {
-          ordering = ordering
-          deleted = deleted
-          persistenceId = persistence_id
-          sequenceNumber = sequence_number
-          created = created
-          tags = tags
-          message = message
-          identifier = identifier
-          manifest = manifest
+        journal {
+          table-name = "journal"
+          columns {
+            ordering = ordering
+            deleted = deleted
+            persistenceId = persistence_id
+            sequenceNumber = sequence_number
+            created = created
+            tags = tags
+            message = message
+            identifier = identifier
+            manifest = manifest
+          }
         }
-        sqlserver-compat-column-names {
-          ordering = Ordering
-          deleted = IsDeleted
-          persistenceId = PersistenceId
-          sequenceNumber = SequenceNr
-          created = Timestamp
-          tags = Tags
-          message = Payload
-          identifier = SerializerId
-          manifest = Manifest
+        
+        metadata {
+          table-name = "journal_metadata"
+          columns {
+            persistenceId = persistenceId
+            sequenceNumber = sequenceNr
+          }
         }
-        sqlite-compat-column-names {
-          ordering = ordering
-          deleted = is_deleted
-          persistenceId = persistence_id
-          sequenceNumber = sequence_nr
-          created = timestamp
-          tags = tags
-          message = payload
-          identifier = serializer_id
-          manifest = manifest
+        
+        tag {
+          table-name = "tags"
+          columns {
+            ordering-id = ordering_id
+            tag-value = tag
+            writer-uuid = writer_uuid
+          }
+        }
+      }
+      
+      # Akka.Persistence.SqlServer compatibility table name and column name mapping
```

Comment:
So if I'm interpreting this correctly: you wanted to make the relationships between column mappings and their respective database engines more explicit and clearer reason about by breaking them up into their own independent / nestable sections, which all tie back into ` table-mapping = [value]` - because in theory, you could replace `value` with something entirely custom and that would work - say, for instance, Oracle support?

Reply:
Exactly. The table mapping is completely decoupled now, it is generic enough that people can add their own entries to an SQL server that we don't even know. It should be safe enough as long as these 3rd party plugins follows our standard table schema.

Reply:
For clarity, this is the database defining HOCON block:
```
database-name {
    schema-name = null
    
    journal {
        table-name = "journal"
        columns {
            ordering = ordering
            deleted = deleted
            persistenceId = persistence_id
            sequenceNumber = sequence_number
            created = created
            tags = tags
            message = message
            identifier = identifier
            manifest = manifest
        }
    }
    
    metadata {
        table-name = "journal_metadata"
        columns {
            persistenceId = persistenceId
            sequenceNumber = sequenceNr
        }
    }
    
    tag {
        table-name = "tags"
        columns {
            ordering-id = ordering_id
            tag-value = tag
            writer-uuid = writer_uuid
        }
    }
}
```

Reply:
> So if I'm interpreting this correctly: you wanted to make the relationships between column mappings and their respective database engines more explicit and clearer reason about by breaking them up into their own independent / nestable sections, which all tie back into ` table-mapping = [value]` - because in theory, you could replace `value` with something entirely custom and that would work - say, for instance, Oracle support?

Well, it -might- be a little more involved than that.

the `table-mapping` is only sets up the compatibility mapping for old tables (otherwise, we use the column names more or less specced by persistence-jdbc). `provider-name` is what actually sets up the per-DB mapping.

I make that disclaimer, because some database providers may or may not require additional mapping to deal with compatibility (main example, [SQLite and PG snapshot timestamps](https://github.com/akkadotnet/Akka.Persistence.Linq2Db/blob/4e48a79e0e6a6c1ac6fbbd29355966e4c4a758d8/src/Akka.Persistence.Sql.Linq2Db/Db/AkkaPersistenceDataConnectionFactory.cs#L102-L107)) 

FWIW most mapping changes should be minor, (most typically around timestamp handling on snapshots if I had to guess.)

Reply:
You have a point, the table column data type themselves are not standardized, some would need adapters to make them work.

Reply:
If I remember correctly, those adapters are actually applied based on the provider name, and not by the compatibility table name, so these changes would not affect those.

Reply:
Correct. The above was mostly clarification on work for future DB support (i.e. we may or may not need to add provider specific mappings in addition to this config.)